docs/proposals/secrets.md

```diff
@@ -0,0 +1,80 @@
+## Motivation
```

Comment:
This should be two separate patches. One with the proposal and the other with the implementation.